src/store/podcasts.ts

```diff
@@ -22,8 +24,29 @@ export const usePodcastStore = defineStore("podcasts", {
       { id: "dev2", name: "dev 2 sandbox" },
       { id: "dev3", name: "dev 3 sandbox" },
     ],
+    readAccessPodcasts: [],
   }),
   getters: {
     getPodcasts: (state: State) => state.podcasts,
+    getReadAccessPodcasts: (state: State) => state.readAccessPodcasts,
+  },
+  actions: {
+    async getReadAccessPodcast() {
+      this.readAccessPodcasts = []
+      const db = getFirestore()
+      this.podcasts.map(async (el) => {
+        try {
+          await getDoc(doc(collection(db, el.id))).then((res) => {
+            if (res) {
+              this.readAccessPodcasts.push(el)
+            }
+          })
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (e: any) {
+          if (e.code === "permission-denied") return e.code
+          else return e
+        }
+      })
+    },
```

Comment:
```suggestion
      async getReadAccessPodcast() {
        const db = getFirestore()
        this.readAccessPodcasts =  this.podcasts.filter((podcast) => {
          try {
            await getDoc(doc(collection(db, el.id)))
            return true
          } catch (e: any) {
            if (e.code === "permission-denied") return false
            else throw(e)
           }
       })
     }
```

Reply:
indentation and syntax is probably is off in suggestion since i wrote in github not an ide :)

Reply:
I have checked your Suggestion in my local but are not working.

Reply:
i wrote it blind, it probably won't work directly, the point was filter rather than map

Reply:
I have used a filter.
please refer -> https://i.imgur.com/Luea2Y9.png